### PR TITLE
undollar: migrate from nodePackages

### PR DESCRIPTION
**Pull request by pyrox0**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/un/undollar/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      --replace-fail "/usr/bin/env node" "${lib.getExe nodejs}"
+  '';
+
+  meta = {
+    description = "eats the dollar sign in the command you just pasted into your terminal";
+    mainProgram = "$";
+    homepage = "https://github.com/xtyrrell/undollar";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ pyrox0 ];
+  };
```
Comment (wegank) [inline pkgs/by-name/un/undollar/package.nix:33]:
```suggestion
    maintainers = with lib.maintainers; [ pyrox0 ];
    inherit (nodejs.meta) platforms;
  };
```

Hopefully, this should remove the `10.rebuild-darwin: 0` label.

File: pkgs/by-name/un/undollar/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenvNoCC,
+  nodejs
+}:
+
+stdenvNoCC.mkDerivation {
+  pname = "undollar";
+  version = "1.0.0";
```
Comment (wegank) [inline pkgs/by-name/un/undollar/package.nix:10]:
```suggestion
  version = "1.0.0-unstable-2018-09-14";
```

File: pkgs/by-name/un/undollar/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    rev = "27e5f0f87ddc4c9b58fe02a68e83a601078ebb43";
+    hash = "sha256-2nudiUh8B5tSg3TeKh1FEJaf8MJ18/IkYikFD07c4Pw=";
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mv undollar.js $out/bin/$
+    substituteInPlace $out/bin/$ \
+      --replace-fail "/usr/bin/env node" "${lib.getExe nodejs}"
+  '';
```
Comment (wegank) [inline pkgs/by-name/un/undollar/package.nix:25]:
```suggestion
  installPhase = ''
    runHook preInstall
    mkdir -p $out/bin
    mv undollar.js $out/bin/$
    substituteInPlace $out/bin/$ \
      --replace-fail "/usr/bin/env node" "${lib.getExe nodejs}"
    runHook postInstall
  '';
```

Comment (pyrox0) [reply pkgs/by-name/un/undollar/package.nix:25]: Fixed

**Pull request merged**